internal/http_helper.go

```diff
@@ -40,20 +40,25 @@ func findContainerRoleByAddress(addr string, request *Request) (*iam.Role, error
 
 	err := backoff.RetryNotify(retryable, b, notify)
 	if err != nil {
-		return nil, err
+		return nil, "", err
 	}
 
 	roleName, err := findDockerContainerIAMRole(container, request)
 	if err != nil {
-		return nil, err
+		return nil, "", err
+	}
+
+	externalId, err := findDockerContainerExternalId(container, request)
```

Comment:
since this can never fail (see the other comment on removing the err) the error handling can just go away here